parquet-avro/src/test/java/org/apache/parquet/avro/TestArrayCompatibility.java

```diff
@@ -653,8 +654,8 @@ public void testAvroCompatRequiredGroupInListWithSchema() throws Exception {
     Path test = writeDirect(
         "message AvroCompatRequiredGroupInListWithSchema {" +
             "  optional group locations (LIST) {" +
-            "    repeated group array {" +
-            "      optional group element {" +
+            "    repeated group bag {" +
+            "      optional group array_element {" +
```

Comment:
According the following `LIST` backwards-compatibility rule

> If the repeated field is a group with one field and is named either `array` or uses the `LIST`-annotated group's name with `_tuple` appended then the repeated type is the element type and elements are required.

the repeated `array` group in the original schema should be interpreted as element type since it has only a single field.  However, the testing data below actually treat it as the repeated wrapper group, and thus fails after `AvroIndexedRecordConverter.isElementType()` is fixed.


According to other code within this test case, this schema was designed to test parquet-hive compatibility.  By running the following SQL statement in Hive 0.13.1

```sql
CREATE TABLE parquet_test
STORED AS PARQUET
AS SELECT ARRAY(
  NAMED_STRUCT("latitude", 1.0, "longitude", 2.0),
  NAMED_STRUCT("latitude", 1.0, "longitude", 2.0)
) AS locations;
```

we may get a similar schema (printed with `parquet-schema`):

```
message hive_schema {
  optional group locations (LIST) {
    repeated group bag {
      optional group array_element {
        optional double latitude;
        optional double longitude;
      }
    }
  }
}
```

That's why I updated this test case.